wave_optimization/CMakeLists.txt

```diff
@@ -1,28 +1,20 @@
 PROJECT(wave_optimization)
 
-# PACKAGES
-FIND_PACKAGE(GTSAM) # optional for now; used only in tests (@todo)
-
-WAVE_ADD_LIBRARY(${PROJECT_NAME}
-    src/ceres/ba.cpp
-    src/ceres/ceres_examples.cpp)
-
-# Use these headers when building, and make clients use them
-WAVE_INCLUDE_DIRECTORIES(${PROJECT_NAME} PUBLIC include)
-
-# Depend on these modules and external libraries, and make clients use them
-TARGET_LINK_LIBRARIES(${PROJECT_NAME} PUBLIC
+WAVE_ADD_MODULE(${PROJECT_NAME} DEPENDS
     wave::utils
     wave::kinematics
     wave::vision
     Eigen3::Eigen
-    ceres)
+    ceres
+    SOURCES
+    src/ceres/ba.cpp
+    src/ceres/ceres_examples.cpp)
```

Comment:
Just curious, do we need these ceres examples in libwave? Like I'm pretty sure they're the examples straight from their website?

Reply:
I thought they were original examples, but yes they don't add much to libwave..

Reply:
they're also the culprit for that whole Eigen/Ceres versioning debacle 😉

Reply:
Yes well we will _actually_ use them soon... I hope